uqcsbot/scripts/link.py

```diff
@@ -0,0 +1,161 @@
+        return (global_match.value, "global") if global_match else (None, None)
+
+    if link_scope == LinkScope.CHANNEL:
+        return (channel_match.value, "channel") if channel_match else (None, None)
+
+    if channel_match:
+        return channel_match.value, "channel"
+
+    if global_match:
+        return global_match.value, "global"
+
+    return None, None
+
+
+@bot.on_command('link')
+@loading_status
+def handle_link(command: Command) -> None:
+    """
+    `!link [-c | -g] [-f] key [value [value ...]]` - Set and retrieve information in a key value
+    store. Links can be set to be channel specific or global. Links are set as global by default,
+    and channel specific links are retrieved by default unless overridden with the respective flag.
+    """
+    parser = ArgumentParser("!link", add_help=False)
+    parser.add_argument("key", type=str, help="Lookup key")
+    parser.add_argument("value", type=str, help="Value to associate with key", nargs="*")
+    flag_group = parser.add_mutually_exclusive_group()
+    flag_group.add_argument("-c", "--channel", action="store_true", dest="channel_flag",
+                            help="Ensure a channel link is retrieved, or none is")
+    flag_group.add_argument("-g", "--global", action="store_true", dest="global_flag",
+                            help="Ignore channel link and force retrieval of global")
+    parser.add_argument("-f", "--force-override", action="store_true", dest="override",
+                        help="Must be passed if overriding a link")
+
+    try:
+        args = parser.parse_args(command.arg.split() if command.has_arg() else [])
+    except SystemExit:
+        # Incorrect Usage
+        return bot.post_message(command.channel_id, "",
+                                attachments=[Attachment(SectionBlock(str(parser.format_help())),
+                                                        color=Color.YELLOW)._resolve()])
+
+    channel = bot.channels.get(command.channel_id)
+    if not channel:
+        return bot.post_message(command.channel_id, "", attachments=[
+            Attachment(SectionBlock("Cannot find channel name, please try again."),
+                       color=Color.YELLOW)._resolve()
+        ])
+    else:
+        channel_name = bot.channels.get(command.channel_id).name
+
+    link_scope = LinkScope.CHANNEL if args.channel_flag else \
+        LinkScope.GLOBAL if args.global_flag else None
+
+    # Retrieve a link
+    if not args.value:
+        link_value, source = get_link_value(key=args.key,
+                                            channel=channel_name,
+                                            link_scope=link_scope)
+        if_channel_flag = f" in channel `{channel_name}`" if args.channel_flag else ""
+        response = f"{args.key} ({source if source == 'global' else channel_name}): " \
+                   f"{link_value}" if link_value else \
+                   f"No link found for key: `{args.key}`{if_channel_flag}"
```

Comment:
A bit of a mess with all the ternaries 😅 Can we split it into something like this instead?
```python
channel_text = f" in channel `{channel_name}`" if args.channel_flag else ""
source_text = source if source == 'global' else channel_name
result_text = link_value if link_value else
    f"No link found for key: `{args.key}`" + channel_text
response = f"{args.key} ({source_text}): {result_text}"
```